### PR TITLE
Harmonize command formats and ensure autocompletion is same

### PR DESCRIPTION
**Pull request by alamirault**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This PR harmonize commands format option description and use sames values than autocompletion.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php [hunk 1]
```diff
@@ -56,9 +59,8 @@ protected function configure(): void
   <info>php %command.full_name% framework</info>
   <info>php %command.full_name% FrameworkBundle</info>
 
-With the <info>--format</info> option specifies the format of the configuration,
-this is either <comment>yaml</comment> or <comment>xml</comment>.
-When the option is not provided, <comment>yaml</comment> is used.
+The <info>--format</info> option specifies the format of the configuration,
+this is either "{$helpFormats}".
```
Comment (alamirault) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: Like `src/Symfony/Bundle/FrameworkBundle/Command/ConfigDebugCommand.php`

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php [hunk 2]
```diff
@@ -41,13 +41,16 @@ class ConfigDumpReferenceCommand extends AbstractConfigCommand
...
     {
+        $commentedHelpFormats = array_map(static fn (string $format): string => sprintf('<comment>%s</comment>', $format), $this->getAvailableFormatOptions());
+        $helpFormats = implode('", "', $commentedHelpFormats);
+
         $this
             ->setDefinition([
                 new InputArgument('name', InputArgument::OPTIONAL, 'The Bundle name or the extension alias'),
                 new InputArgument('path', InputArgument::OPTIONAL, 'The configuration option path'),
-                new InputOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (yaml or xml)', 'yaml'),
+                new InputOption('format', null, InputOption::VALUE_REQUIRED, sprintf('The output format ("%s")', implode('", "', $this->getAvailableFormatOptions())), class_exists(Yaml::class) ? 'yaml' : 'json'),
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: You change the default value when Yaml is not available? Smart 👍🏻

**Review by GromNaN — APPROVED** [review]
✅ That will be less code to update and less risk of missing something when an output format is added.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: I'm not so sure. I would prefer to have a default that is "stable". In any case, it's not related to this PR, so let's revert and discuss the default change elsewhere.

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php [hunk 3]
```diff
@@ -352,4 +351,11 @@ public function filterToServiceTypes(string $serviceId): bool
 
         return class_exists($serviceId) || interface_exists($serviceId, false);
     }
+
+    private function getAvailableFormatOptions(): array
+    {
+        $helper = new DescriptorHelper();
+
+        return $helper->getFormats();
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php]: Could be a one liner.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php (hunk 2 above)
Comment (alamirault) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: I agree, it's not related to this PR.

(FYI ConfigDebugCommand has this behavior)

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: `either` is wrong here as you removed `or` after that in favor of a comma.

File: src/Symfony/Bundle/FrameworkBundle/Command/EventDispatcherDebugCommand.php [hunk 4]
```diff
@@ -155,4 +155,11 @@ private function searchForEvent(EventDispatcherInterface $dispatcher, string $ne
 
         return $output;
     }
+
+    private function getAvailableFormatOptions(): array
+    {
+        $helper = new DescriptorHelper();
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Command/EventDispatcherDebugCommand.php]: you should inline it, as done in ContainerDebugCommand and RouterDebugCommand.

File: src/Symfony/Component/Uid/Command/GenerateUlidCommand.php [hunk 5]
```diff
@@ -110,7 +104,16 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
     {
         if ($input->mustSuggestOptionValuesFor('format')) {
-            $suggestions->suggestValues(self::FORMAT_OPTIONS);
+            $suggestions->suggestValues($this->getAvailableFormatOptions());
```
Comment (stof) [inline src/Symfony/Component/Uid/Command/GenerateUlidCommand.php:107]: you could keep the constant actually.

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php (hunk 1 above)
Comment (alamirault) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: Can you tell me what is the best here ?

(I'm not native speaker)

File: src/Symfony/Component/Uid/Command/GenerateUlidCommand.php (hunk 5 above)
Comment (alamirault) [reply src/Symfony/Component/Uid/Command/GenerateUlidCommand.php:107]: `GenerateUuidCommand` command don't use constant. I think is better if theses two commands have same structure. (And same like other commands)

File: src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php (hunk 1 above)
Comment (alexisLefebvre) [inline src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]:
```suggestion
these are "{$helpFormats}".
```

Comment (alamirault) [reply src/Symfony/Bundle/FrameworkBundle/Command/ConfigDumpReferenceCommand.php]: Thanks ! Applied also on `ConfigDebugCommand`

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alamirault.

**Pull request merged**